### PR TITLE
GO-1777: add simple nested query filter

### PR DESCRIPTION
**Pull request by deff7**

### CODE REVIEW

File: pkg/lib/database/filter.go [hunk 1]
```diff
@@ -16,8 +18,10 @@ var (
 	ErrValueMustBeListSupporting = errors.New("value must be list supporting")
 )
 
-func MakeAndFilter(protoFilters []*model.BlockContentDataviewFilter, store OptionsGetter) (AndFilters, error) {
-
+func MakeAndFilter(protoFilters []*model.BlockContentDataviewFilter, store ObjectStore) (AndFilters, error) {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/database/filter.go:21]: This name of the function is strange to me :(

File: pkg/lib/database/filter.go [hunk 2]
```diff
@@ -33,135 +37,168 @@ func MakeAndFilter(protoFilters []*model.BlockContentDataviewFilter, store Optio
...
 }
 
-func MakeFilter(proto *model.BlockContentDataviewFilter, store OptionsGetter) (Filter, error) {
+func MakeFilter(rawFilter *model.BlockContentDataviewFilter, store ObjectStore) (Filter, error) {
+	if store == nil {
+		return AndFilters{}, fmt.Errorf("objectStore dependency is nil")
+	}
+	parts := strings.SplitN(rawFilter.RelationKey, ".", 2)
+	if len(parts) == 2 {
+		rawNestedFilter := proto.Clone(rawFilter).(*model.BlockContentDataviewFilter)
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/database/filter.go:46]: May be extract to separate function?

File: pkg/lib/database/filter.go [hunk 3]
```diff
@@ -201,6 +252,18 @@ func (a OrFilters) String() string {
 	return fmt.Sprintf("(%s)", strings.Join(orS, " OR "))
 }
 
+func (a OrFilters) IterateNestedFilters(fn func(nestedFilter Filter) error) error {
+	for _, f := range a {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/database/filter.go:256]: It's seems like duplicated logic to 
```
func (a AndFilters) IterateNestedFilters(fn func(nestedFilter Filter) error) error {
	for _, f := range a {
		if withNested, ok := f.(WithNestedFilter); ok {
			err := withNested.IterateNestedFilters(fn)
			if err != nil {
				return err
			}
		}
	}
	return nil
}
```

File: pkg/lib/database/filter.go (hunk 1 above)
Comment (fat-fellow) [reply pkg/lib/database/filter.go:21]: the same issue, but do you have any idea how to make it batter?
possibly MakeFilterAnd can sound slightly better

Comment (AnastasiaShemyakinskaya) [reply pkg/lib/database/filter.go:21]: Yes, sounds better. Probably BuildFilterAnd

Comment (deff7) [reply pkg/lib/database/filter.go]: Thank you, I renamed AndFilter and OrFilter to `FiltersAnd` and `FiltersOr`, think it's less confusing. Also added prefix `Filter` for all filters

Comment (deff7) [reply pkg/lib/database/filter.go]: Now it's `MakeFiltersAnd`

File: pkg/lib/database/filter.go (hunk 2 above)
Comment (deff7) [reply pkg/lib/database/filter.go]: done!

File: pkg/lib/database/filter.go (hunk 3 above)
Comment (deff7) [reply pkg/lib/database/filter.go]: extracted, thank you

**Review by fat-fellow — APPROVED** [review]

**Pull request merged**